### PR TITLE
GH-115976: Add WASI to CI

### PR DESCRIPTION
**Pull request by brettcannon**
* Issue: gh-115976

### CODE REVIEW

**Review by sobolevn — COMMENTED** [review]
Please, take a look that tests pass but produce this warning:

```
 z stack-size=524288 -Wl,--stack-first -Wl,--initial-memory=10485760   Modules/_testexternalinspection.o   -o Modules/_testexternalinspection.cpython-313d-wasm32-wasi.so
/bin/sh: 1: z: not found
make: [Makefile:3362: Modules/_testexternalinspection.cpython-313d-wasm32-wasi.so] Error 127 (ignored)
Error:  _testexternalinspection (/home/runner/work/cpython/cpython/cross-build/wasm32-wasi/build/lib.wasi-wasm32-3.13-pydebug/_testexternalinspection.cpython-313d-wasm32-wasi.so) is missing
```

**Comment by hugovk** [comment]
`build_wasi` looks a bit lonely at https://github.com/python/cpython/actions/runs/8208735060?pr=116516 :

image

I think it requires listing in the `needs` section:

https://github.com/python/cpython/blob/c951e25c24910064a4c8b7959e2f0f7c0d4d0a63/.github/workflows/build.yml#L519-L532

And then in either `allowed-failures` (if allowed to fail when selected to run) or `allowed-skips` (if it must pass when selected to run):

https://github.com/python/cpython/blob/c951e25c24910064a4c8b7959e2f0f7c0d4d0a63/.github/workflows/build.yml#L540-L580

**Comment by brettcannon** [comment]
> Please, take a look that tests pass but produce this warning:
> 
> ```
>  z stack-size=524288 -Wl,--stack-first -Wl,--initial-memory=10485760   Modules/_testexternalinspection.o   -o Modules/_testexternalinspection.cpython-313d-wasm32-wasi.so
> /bin/sh: 1: z: not found
> make: [Makefile:3362: Modules/_testexternalinspection.cpython-313d-wasm32-wasi.so] Error 127 (ignored)
> Error:  _testexternalinspection (/home/runner/work/cpython/cpython/cross-build/wasm32-wasi/build/lib.wasi-wasm32-3.13-pydebug/_testexternalinspection.cpython-313d-wasm32-wasi.so) is missing
> ```

https://github.com/python/cpython/issues/116229

**Comment by brettcannon** [comment]
> `build_wasi` looks a bit lonely

Fixed!

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -527,6 +535,7 @@ jobs:
     - build_ubuntu_ssltests
     - build_windows
     - build_windows_free_threading
+    - build_wasi
```
Comment (hugovk) [inline .github/workflows/build.yml]:
```suggestion
    - build_wasi
    - build_windows
    - build_windows_free_threading
```

**Review by hugovk — APPROVED** [review]

**Pull request merged**